openbb_platform/providers/fmp/openbb_fmp/models/form_13FHR.py

```diff
@@ -0,0 +1,138 @@
+    symbol: Optional[str] = Field(
+        default=None, description=DATA_DESCRIPTIONS.get("symbol", "")
+    )
+    filling_date: Optional[dateType] = Field(
+        default=None, description="Date when the filing was submitted to the SEC."
+    )
+    accepted_date: Optional[dateType] = Field(
+        default=None, description="Date when the filing was accepted by the SEC."
+    )
+    link: Optional[str] = Field(
+        default=None, description="URL link to the SEC filing on the SEC website."
+    )
+    final_link: Optional[str] = Field(
+        default=None,
+        description="URL link to the XML information table of the SEC filing.",
+    )
+
+
+class FMPForm13FHRFetcher(
+    Fetcher[
+        FMPForm13FHRQueryParams,
+        List[FMPForm13FHRData],
+    ]
+):
+    """Fetches and transforms data from the Form 13f endpoints."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FMPForm13FHRQueryParams:
+        """Transform the query params."""
+        return FMPForm13FHRQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: FMPForm13FHRQueryParams,
+        credentials: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the Form 13f endpoint."""
+        api_key = credentials.get("fmp_api_key") if credentials else ""
+        if query.symbol.isnumeric():
+            cik = query.symbol
+        else:
+            try:
+                profile = await FMPEquityProfileFetcher.fetch_data({"symbol": query.symbol}, {"fmp_api_key": api_key})
+                cik = getattr(profile[0], "cik", "")
+                if not cik:
+                    raise OpenBBError(f"Invalid symbol -> {query.symbol}")
+            except OpenBBError as e:
+                raise OpenBBError(f"Invalid symbol -> {query.symbol} -> {e}")
+        date_url = create_url(
+            3,
+            f"form-thirteen-date/{cik}",
+            api_key,
+            query,
+            exclude=["symbol", "limit"],
+        )
+        dates = await amake_request(date_url, response_callback=response_callback, **kwargs)
+        if not dates:
+            raise EmptyDataError("No data returned for the given symbol.")
+        if not dates or len(dates) == 0:
+            warn(f"Symbol Error: No data found for symbol {cik}")
+        results: List[Dict] = []
+
+        async def get_one(date):
+            """Get data for the given date."""
+            date = datetime.strptime(date, "%Y-%m-%d").date()
+            query.date = date
+            url = create_url(
+                3,
+                f"form-thirteen/{cik}",
+                api_key,
+                query,
+                exclude=["symbol", "limit"],
+            )
+            result = await amake_request(url, response_callback=response_callback, **kwargs)
+            if not result or len(result) == 0:
+                warn(f"Symbol Error: No data found for symbol {symbol}")
```

Comment:
Undefined local here.
```suggestion
                warn(f"Symbol Error: No data found for symbol {query.symbol}")
```